src/main/java/it/gov/pagopa/payhub/activities/service/ingestionflow/IngestionFileHandlerService.java

```diff
@@ -0,0 +1,71 @@
+package it.gov.pagopa.payhub.activities.service.ingestionflow;
+
+import it.gov.pagopa.payhub.activities.util.AESUtils;
+import it.gov.pagopa.payhub.activities.util.FileUtils;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * Service class responsible for handling encrypted ingestion files,
+ * including decryption, validation, and extraction of ZIP files.
+ */
+@Slf4j
+@Service
+public class IngestionFileHandlerService {
+	private static final String TEMPORARY_PATH = "TEMP";
```

Comment:
This way you are referencing a local directory, not the tmp directory...
The local directory probably it will be the share.

maybe it's better to use the /tmp/ directory of the running pod in order to avoid to store in the share personal data
(you have to create a folder based on the input file, maybe under this folder you could create the same folder hierarchy of the current file)